.github/workflows/mcad-CI.yml

```diff
@@ -0,0 +1,37 @@
+# This workflow will build mcad and run unit test and E2E tests
+name: MCAD-CI
+on:
+  pull_request_target:
```

Comment:
Is there a reason to use `pull_request_target`?
Isn't it enough to use `pull_request`?

AFAIK `pull_request_target` is needed in case you need write access to repository or access to GitHub secrets.

Reply:
yes `pull_request` is safer, updated to the same